environment.go

```diff
@@ -979,13 +978,25 @@ func getClientConfig(baseURL string) (*ClientConfig, error) {
 		parts := strings.Split(path, string(filepath.Separator))
 		buildInfo := strings.Split(parts[len(parts)-2], "_")
 		displayName := ValidArchDisplayNames[fmt.Sprintf("%s_%s", buildInfo[0], buildInfo[1])]
+		archPath := strings.Join(parts[len(parts)-2:], "/")
 		c.ClientBinaries = append(c.ClientBinaries, ClientBinary{
-			URL: fmt.Sprintf("%s/%s/%s", baseURL, ClientDirectory,
-				strings.Join(parts[len(parts)-2:], "/")),
+			URL:         fmt.Sprintf("%s/%s/%s", baseURL, ClientDirectory, archPath),
 			OS:          buildInfo[0],
 			Arch:        buildInfo[1],
 			DisplayName: displayName,
 		})
+		if s3BaseURL != "" {
+			c.S3ClientBinaries = append(c.S3ClientBinaries, ClientBinary{
```

Comment:
The UI uses the ClientBinaries field for what it presents on the settings page. Are you intentionally hiding the ones from S3 for now?

Reply:
That was more for backwards compability  so that the old self-updater can update without getting it from S3. The old logic only checks one URL, so if it's the S3 URL and the S3 URL is broken, it'll fail to self-update. The new field is to gracefully handle the self-updater's update.

I feel like most people care about self-updating rather than downloading the binary for the first time. We could add more URLs for S3 downloads for first-time downloaders, but I think it's not critical for now (and I don't want to mess with Spruce UI stuff).